api/src/main/java/org/pmiops/workbench/api/UserController.java

```diff
@@ -135,19 +139,46 @@ public UserController(
     return ResponseEntity.ok(response);
   }
 
-  private BillingAccount freeTierBillingAccount() {
-    return new BillingAccount()
-        .isFreeTier(true)
-        .displayName("Use All of Us free credits")
-        .name(configProvider.get().billing.freeTierBillingAccountName())
-        .isOpen(true);
-  }
-
   @Override
   public ResponseEntity<WorkbenchListBillingAccountsResponse> listBillingAccounts() {
-    // this check should be redundant: the UI only calls this endpoint when the flag is active
+    List<BillingAccount> billingAccounts =
+        Stream.concat(maybeFreeTierBillingAccount(), maybeCloudBillingAccounts())
+            .collect(Collectors.toList());
+
+    return ResponseEntity.ok(
+        new WorkbenchListBillingAccountsResponse().billingAccounts(billingAccounts));
+  }
+
+  /**
+   * The logic here is a little weird. See RW-4857.
+   *
+   * <p>When Billing Upgrade is true: return the Free Tier account only when the user has remaining
```

Comment:
Note: This is for another day, but there's an argument to always return it, but just have the option be disabled in the dropdown if credits are expired. Driving business logic here based on the caller's free credits is dubious per prior discussions (even if we're disallowing certain interactions client-side or with validation in other methods).